benchmark/simulate_hgts.py

```diff
@@ -0,0 +1,627 @@
+    """Launch OrthoFinder to report orthologous gene groups for two genomes.
+
+    Parameters
+    ----------
+    proteomes_dir: string
+        directory path storing FASTA coding sequences for complete genomes
+    threads: integer
+        number of threads to use
+    verbose: boolean
+        if True, run in verbose mode
+    """
+    if verbose:
+        sys.stdout.write("\tLaunch OrthoFinder ..\n")
+    orthofinder_command = ["orthofinder.py",
+                           "-f", proteomes_dir,
+                           "-t", str(threads)]
+    proc = subprocess.Popen(orthofinder_command,
+                            stdout=subprocess.PIPE,
+                            stderr=subprocess.PIPE,
+                            close_fds=True)
+    proc.wait()
+    stdout, stderr = proc.communicate()
+    if stderr:
+        print(stderr)
+    if verbose:
+        print(stdout)
+        sys.stdout.write("\tDone\n")
+
+
+def parse_orthofinder(results_dir):
+    """Parse the output files of OrthoFinder for orthologous genes.
+
+    Parameters
+    ----------
+    results_dir: string
+        OrthoFinder results directory
+
+    Returns
+    -------
+    species_ids: dictionary
+    sequence_ids: dictionary
+    orthologous_groups: list of lists
+    """
+    species_ids = {}
+    sequence_ids = {}
+    orthologous_groups = []
+    with open(
+        glob.glob(
+            join(
+                results_dir,
+                "clusters_OrthoFinder_*_id_pairs.txt"))[0], 'r') as id_pairs_f:
+        # skip header lines
+        for _ in range(7):
+            next(id_pairs_f)
+        # parse orthologous family groups
+        for line in id_pairs_f:
+            line = line.strip().split()
+            # include only families with at least 2 orthologs
+            if len(line[1:-1]) > 1:
+                orthologous_groups.append(line[1:-1])
+    with open(join(results_dir, "SpeciesIDs.txt"), 'r') as species_ids_f:
+        for line in species_ids_f:
+            line = line.strip().split()
+            species_ids[line[0].split(':')[0]] = line[1]
+    with open(join(results_dir, "SequenceIDs.txt"), 'r') as sequence_ids_f:
+        for line in sequence_ids_f:
+            line = line.strip().split()
+            sequence_ids[line[0].split(':')[0]] = line[1]
+
+    return species_ids, sequence_ids, orthologous_groups
+
+
+def simulate_orthologous_rep(genes_donor,
+                             seq_donor,
+                             genes_recip,
+                             seq_recip,
+                             sequence_ids,
+                             orthologous_groups,
+                             orthologous_rep_prob,
+                             percentage_hgts,
+                             log_f):
+    """Simulate orthologous replacement HGT.
+
+    Notes
+    -----
+    Algorithm: Using list of orthologous genes between donor and recipient
+    genomes
+
+    Parameters
+    ----------
+    genes_donor: dictionary
```

Comment:
If possible, I think it would be useful to have 1 sentence descriptions of the parameters since it looks like this is a public method